### PR TITLE
Add test discovery and NoBuild option to test requests

### PR DESCRIPTION
**Pull request by Samirat**
More info about this in the related VSCode extension PR:
https://github.com/OmniSharp/omnisharp-vscode/pull/3601

This adds
1. A DiscoverTests request, which lets you get info for all tests in an assembly
2. A NoBuild option for test requests, which lets you run/debug tests without always building first

There is a [separate PR](https://github.com/OmniSharp/omnisharp-roslyn/pull/1710) with just the runsettings additions.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "lockfileVersion": 1
```
Comment (JoeRobich) [inline package-lock.json:2]: Why is this file necessary?

Comment (Samirat) [reply package-lock.json]: Not sure how that got there, removed.

**Comment by JoeRobich** [comment]
@Samirat Thanks!
